fenix/app/src/main/java/org/mozilla/fenix/debugsettings/tabs/TabTools.kt

```diff
@@ -322,46 +336,34 @@ private class TabToolsPreviewParameterProvider : PreviewParameterProvider<TabToo
 private fun TabToolsPreview(
     @PreviewParameter(TabToolsPreviewParameterProvider::class) model: TabToolsPreviewModel,
 ) {
-    val snackbarState = remember { SnackbarHostState() }
-    val scope = rememberCoroutineScope()
-
-    with(model) {
-        FirefoxTheme {
-            Box(
-                modifier = Modifier.background(color = FirefoxTheme.colors.layer1),
-            ) {
-                TabToolsContent(
-                    activeTabCount = activeTabCount,
-                    inactiveTabCount = inactiveTabCount,
-                    privateTabCount = privateTabCount,
-                    totalTabCount = totalTabCount,
-                    inactiveTabsEnabled = inactiveTabsEnabled,
-                    onAddTabsToActiveClick = {
-                        scope.launch {
-                            snackbarState.showSnackbar("Add $it tabs to active")
-                        }
-                    },
-                    onAddTabsToInactiveClick = {
-                        scope.launch {
-                            snackbarState.showSnackbar("Add $it tabs to inactive")
-                        }
-                    },
-                    onAddTabsToPrivateClick = {
-                        scope.launch {
-                            snackbarState.showSnackbar("Add $it tabs to private")
-                        }
-                    },
-                )
+    var activeTabCount by remember { mutableIntStateOf(model.activeTabCount) }
+    var inactiveTabCount by remember { mutableIntStateOf(model.inactiveTabCount) }
+    var privateTabCount by remember { mutableIntStateOf(model.privateTabCount) }
+    val totalTabCount = remember(activeTabCount, inactiveTabCount, privateTabCount) {
+        activeTabCount + inactiveTabCount + privateTabCount
+    }
 
-                SnackbarHost(
-                    hostState = snackbarState,
-                    modifier = Modifier.align(Alignment.BottomCenter),
-                ) { snackbarData ->
-                    Snackbar(
-                        snackbarData = snackbarData,
-                    )
-                }
-            }
+    // todo: is it better to have this (and be framework agnostic), or to be able to validate the Store logic in a Compose preview?
+    // todo: just make a separate preview for the above?
```

Comment:
these are two open questions of how we should think about constructing previews going forward, and striking the right balance of testing the UI versus testing the business logic WITH the UI.